org.eclipse.xtext.ide/src/org/eclipse/xtext/ide/server/LanguageServerImpl.java

```diff
@@ -232,18 +231,25 @@ public CompletableFuture<InitializeResult> initialize(InitializeParams params) {
 		result.setCapabilities(createServerCapabilities(params));
 		access.addBuildListener(this);
 		return requestManager.runWrite(() -> {
-			if (workspaceManager.isSupportsWorkspaceFolders()) {
+			if (isWorkspaceFoldersSupported() && workspaceManager.isSupportsWorkspaceFolders()) {
 				List<WorkspaceFolder> workspaceFolders = params.getWorkspaceFolders();
 				if (workspaceFolders == null)
 					workspaceFolders = Collections.emptyList();
 				workspaceManager.initialize(workspaceFolders, this::publishDiagnostics, CancelIndicator.NullImpl);
 			} else {
+				URI baseDir = getBaseDir(params);
 				workspaceManager.initialize(baseDir, this::publishDiagnostics, CancelIndicator.NullImpl);
 			}
 			return result;
 		}, (cancelIndicator, it) -> it).thenApply(it -> initializeResult = it);
 	}
 
+	protected boolean isWorkspaceFoldersSupported() {
```

Comment:
This is in fact about the client capabilities. Can this method be renamed to `clientSupportsWorkspaceFolders`?